lib/spack/spack/solver/asp.py

```diff
@@ -3384,6 +3384,19 @@ def _is_reusable(spec: spack.spec.Spec, packages, local: bool) -> bool:
     return False
 
 
+def _has_runtime_dependencies(spec: spack.spec.Spec) -> bool:
+    if not WITH_RUNTIME:
+        return True
+
+    if spec.satisfies("%gcc") and not spec.satisfies("^gcc-runtime"):
+        return False
+
+    if spec.satisfies("%oneapi") and not spec.satisfies("^intel-oneapi-runtime"):
+        return False
```

Comment:
We might want to generalize this logic, and avoid hard-coding the current runtimes. This should be enough though to start testing the ASP logic, and see if it needs adjustments.